compiler/src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -30,32 +32,38 @@ import Reporter._
 trait Reporting { this: Context =>
 
   /** For sending messages that are printed only if -verbose is set */
-  def inform(msg: => String, pos: SourcePosition = NoSourcePosition): Unit =
+  def inform(msg: => String, pos: Position = NoPosition): Unit =
     if (this.settings.verbose.value) this.echo(msg, pos)
 
-  def echo(msg: => String, pos: SourcePosition = NoSourcePosition): Unit =
-    reporter.report(new Info(msg, pos))
+  def echo(msg: => String, pos: Position = NoPosition): Unit =
+    reporter.report(new Info(msg, sourcePos(pos)))
 
   def reportWarning(warning: Warning): Unit =
     if (!this.settings.silentWarnings.value) {
       if (this.settings.XfatalWarnings.value) reporter.report(warning.toError)
       else  reporter.report(warning)
     }
 
-  def deprecationWarning(msg: => Message, pos: SourcePosition = NoSourcePosition): Unit =
-    reportWarning(new DeprecationWarning(msg, pos))
+  def deprecationWarning(msg: => Message, pos: SourcePosition): Unit = reportWarning(new DeprecationWarning(msg, pos))
+  def deprecationWarning(msg: => Message, pos: Position = NoPosition): Unit = deprecationWarning(msg, sourcePos(pos))
```

Comment:
Why duplicate `deprecationWarning` (and the others like it) but not `inform` and `echo`?'

Overall I am skeptical about this commit. `SourcePosition` is the better type, whereas `Position` is a compromise to gain speed. So it looks better to define the API of reporters in terms of `SourcePosition`. Their usage becomes more flexible that way.  It seems we can get this only at the price of duplicating all methods. Not terrible but not very nice either. The implicit conversion avoids all that boilerplate.